ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/JdbcLedgerDao.scala

```diff
@@ -665,8 +863,49 @@ private class JdbcLedgerDao(
       divulgedContracts: Iterable[DivulgedContract],
       blindingInfo: Option[BlindingInfo],
       recordTime: Instant,
-  )(implicit loggingContext: LoggingContext): Future[PersistenceResponse] =
-    throw new UnsupportedOperationException("not supported")
+  )(implicit loggingContext: LoggingContext): Future[PersistenceResponse] = {
+    logger.info("Storing transaction")
+    dbDispatcher
+      .executeSql(metrics.daml.index.db.storeTransactionDbMetrics) { implicit conn =>
+        sequentialIndexer.store(
+          conn,
+          offset,
+          validate(ledgerEffectiveTime, transaction, divulgedContracts) match {
+            case None =>
+              Some(
+                Update.TransactionAccepted(
+                  optSubmitterInfo = submitterInfo,
+                  transactionMeta = TransactionMeta(
```

Comment:
Could you write a TODO to introduce a light-weight TransactionMeta for the read-path. This heavy one only makes sense in the write path. Using a type in such half-filled manner can lead to bad problems.

Reply:
Well I would not implement an ADT instead the TransactionMeta, because that would obfuscate accessor code all along the lines. First idea comes in mind: just encapsulating this "light" nature with a well placed factory function :thinking: 
Anyhow I think it is good to keep in mind that all this is temporary until removal of sandbox-classic, so it would be great to minimize change on code which is permanent in this sense.